src/analysis/special_functions/log.lean

```diff
@@ -196,6 +196,19 @@ begin
   rw exp_log hx,
 end
 
+/-- Bound for `| log x * x |` in the interval `(0, 1]`. -/
+lemma abs_log_mul_self_lt (x: ℝ) (hx : 0 < x ∧ x ≤ 1) : | log x * x | < 1 :=
```

Comment:
Either use two separate assumptions, or formulate it with `Ioc` (I'd go for the two separate assumptions personnally). Also, in general we don't put space around the absolute values, i.e., we would write it as `|log x * x|` (here and elsewhere in the PR).